analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs

```diff
@@ -94,8 +93,7 @@ private static void CheckInvalidProperties(INamedTypeSymbol parameterType, Sonar
         && (type.Constructors.Any(x => x.Parameters.Length == 0)        // The type must have a parameterless constructor, unless
             || type.IsValueType                                         // - it's a value type
             || type.IsRecord()                                          // - it's a record type
-            || type.IsInterface()                                       // - it's an interface (although the type that implements will be actually used)
-            || type.Is(KnownType.System_String));                       // - it has a custom Model Binder (e.g. System.String has one)
```

Comment:
The [issue title ](https://github.com/SonarSource/sonar-dotnet/issues/9284) is somewhat misleading. It not only refers to strings, but to all non-nullable reference types.
To make this work, the following statement must be removed from [this line](https://github.com/SonarSource/sonar-dotnet/blob/3fea67efaa8e1225b0c92be9ad692d89855075a5/analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs#L100) (that checks for non-nullable reference types):
```cs
type.NullableAnnotation() != NullableAnnotation.NotAnnotated
```